core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -102,19 +94,21 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (!mTachyonFS.requestSpace(length)) {
-      mCanWrite = false;
-
-      String msg =
-          "Local tachyon worker does not have enough " + "space (" + length + ") or no worker for "
-              + mFile.mFileId + " " + mBlockId;
-
-      throw new IOException(msg);
+    if (mAvailableBytes < length) {
```

Comment:
mAvailableBytes records available space bytes for the block, when user successfully request space from worker mAvailableBytes increases, when user writes some data into the file,  mAvailableBytes reduces.